mlir/lib/Conversion/RockToGPU/RockToGPU.cpp

```diff
@@ -330,6 +323,76 @@ void LowerRockOpsToGPUPass::runOnOperation() {
       signalPassFailure();
   });
 
+  // Post processing the created gpuFuncs
+  op.walk([](gpu::GPUFuncOp gpuFunc) {
+    // Calculate lds usage
+    int64_t ldsUsage = 0;
+    for (const auto &en : llvm::enumerate(gpuFunc.getWorkgroupAttributions())) {
+      BlockArgument ldsBuf = en.value();
+      ShapedType ldsBufType = ldsBuf.getType().cast<ShapedType>();
+      Type elemType = ldsBufType.getElementType();
+      int64_t numElems = ldsBufType.getNumElements();
+      int64_t sizeBytes = (elemType.getIntOrFloatBitWidth() / 8) * numElems;
+      ldsUsage += sizeBytes;
+    }
+    OpBuilder b(gpuFunc.getContext());
+    // The following attribute is set to be used by check-residency pass
+    // later on.
+    if (ldsUsage != 0) {
+      gpuFunc->setAttr("rock.shared_buffer_size",
+                       b.getI32IntegerAttr(ldsUsage));
+    }
+    LLVM_DEBUG(llvm::dbgs() << "Attempting to set wavesPerEU...\n");
+    if (!gpuFunc->hasAttrOfType<IntegerAttr>("block_size")) {
+      LLVM_DEBUG(llvm::dbgs() << "blockSize not found in gpuFunc.\n");
+      return;
+    }
+    int64_t blockSize =
+        gpuFunc->getAttrOfType<IntegerAttr>("block_size").getInt();
+    if (!gpuFunc->hasAttrOfType<IntegerAttr>("grid_size")) {
+      LLVM_DEBUG(llvm::dbgs() << "gridSize not found in gpuFunc.\n");
+      return;
+    }
+    int64_t gridSize =
+        gpuFunc->getAttrOfType<IntegerAttr>("grid_size").getInt();
+    StringAttr arch = rock::getArch(gpuFunc);
+    if (arch) {
+      rock::AmdArchInfo archInfo = rock::lookupArchInfo(arch);
+      int64_t wavesPerBlock =
+          blockSize / (archInfo.numEUPerCU * archInfo.waveSize);
+      int64_t wgsPerCU = gridSize / archInfo.minNumCU;
```

Comment:
Is this assuming all WGs resident?

I think we should also have a `rock::getNumCUs(op)` lookup method (like `rock::getArch(op)`).

Reply:
> Is this assuming all WGs resident?

No if WGs are resident, then gridSize / archInfo.minNumCU would be < 1 (zero).

> I think we should also have a rock::getNumCUs(op) lookup method (like rock::getArch(op)).

I disagree. numCUs are not a property of the op rather it is a property of the arch.

Reply:
The arch only specifies the minimum CUs, see numCU on pretty much all the rock.ops. MIGraphX will tell us the actual num_cus and it will be added as a property along with arch.

Reply:
Yeah, numCU's isn't an architectural property.
Remember, `gfx1100` covers multiple different models of Navi31, which have different core counts.

Reply:
Done.